server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -374,7 +374,7 @@ static Condition.Stats buildStats(@Nullable final IndexMetadata metadata, @Nulla
                 .flatMap(Arrays::stream)
                 .filter(shard -> shard.getShardRouting().primary())
                 .map(ShardStats::getStats)
-                .mapToLong(shard -> shard.docs.getCount())
+                .mapToLong(shard -> shard.docs == null ? 0L : shard.docs.getCount())
```

Comment:
Are these changes (this one and the one a couple of lines higher) intentional? I see they're also on `main`, so I _think_ they're ok, but they're not part of your original PR.

Reply:
That is true, I intentionally chose to leave them because it seems to me like a bug fix than a feature.